Tools/CMake/AMReXGenerateConfigHeader.cmake

```diff
@@ -43,6 +43,7 @@ function ( generate_config_header )
            set(COMPILER_ID_MACRO  __PGI)
        elseif ( ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" )       OR
                 ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang" )  OR
+                ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "ROCMClang" )   OR
```

Comment:
Introduced in 3.21.0, this will be fixed in 3.21.3+:
https://gitlab.kitware.com/cmake/cmake/-/merge_requests/6533